app/models/users/notification_setting.rb

```diff
@@ -4,7 +4,21 @@ class NotificationSetting < ApplicationRecord
 
     belongs_to :user, touch: true
 
-    validates :email_digest_periodic, inclusion: { in: [true, false] }
+    validates :email_badge_notifications, exclusion: [nil]
```

Comment:
I'm still not super convinced by the need for these validations, as we ensure database integrity via non-null constraints and the UI should never generate a `nil` here. But as you said, we can decide on this later, it's ok to leave it in for now.

Reply:
> the UI should never generate a nil here.

When I read this, I have finally understood what I felt something wrong.

I was guessing these sending nil tests below are for a case that a user sends nil deliberately by rewriting HTML or doing something like that.
https://github.com/forem/forem/blob/1253d90114c0c795b36de31d10ab2476b716125d/spec/requests/user/user_notification_settings_spec.rb#L40-L44
https://github.com/forem/forem/blob/1253d90114c0c795b36de31d10ab2476b716125d/spec/requests/user/user_notification_settings_spec.rb#L85-L91

As you said, a user never sends nil usually, so I think it would be better to remove these tests.
And do so, we could also remove all the validations on the model.

What do you think?

Reply:
Ah, I think this is where we previously didn't quite manage to get on the same page 😃 I always approached this from the point of our UI where this is a checkbox:

![image](https://user-images.githubusercontent.com/47985/132794120-e03e4013-9b9c-49d8-9abc-f525a14acae3.png)

The way Rails handles these is via a hidden field with a value of "0", so an unchecked box also won't result in `nil`. 

So yes, I think we can remove these specs and the newly added validations, if someone really manages to send a `nil` here I'd be ok with the exception from the DB bubbling up because it would truly be an exceptional circumstance.

Reply:
>  I think we can remove these specs and the newly added validations

I removed the specs and the validations I added.
(I think email_digest_periodic validation is also unnecessary.)

> if someone really manages to send a nil here I'd be ok with the exception from the DB bubbling up because it would truly be an exceptional circumstance.

I totally agree with you 👍